lib/di/provider/dio_provider.dart

```diff
@@ -17,10 +18,9 @@ class DioProvider {
 
   Dio _createDio({bool requireAuthentication = false}) {
     final dio = Dio();
-    final appInterceptor = AppInterceptor(
-      requireAuthentication,
-      dio,
-    );
+    final SecureStorage secureStorage = SecureStorage();
+    final appInterceptor =
+        AppInterceptor(requireAuthentication, dio, secureStorage);
```

Comment:
I prefer this style to improve readability

```suggestion
    final appInterceptor = AppInterceptor(
      requireAuthentication,
      dio,
      secureStorage,
    );
```